src/main/java/org/spongepowered/api/entity/ai/task/AITaskType.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.ai.task;
+
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.entity.living.Agent;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+import java.util.Optional;
+
+@CatalogedBy(AITaskTypes.class)
+public interface AITaskType extends CatalogType {
+
+    /**
+     * Gets the {@link Agent} class that this type is owned by.
+     *
+     * @return The owner class
+     */
+    Optional<Class<? extends Agent>> getOwnerClass();
```

Comment:
Wrong. Plugins don't implement AITaskType, SpongeCommon does this. Due to mods, the owner class is not known until its added to the Entity UNLESS the AITaskType is created by SpongeAPI through the register method. Since I force you to give me the class and I enforce that class to be the same as the generic or a child, it is safe.